cli.js

```diff
@@ -904,9 +912,161 @@ async function execute(network, action, ...params) {
     );
     return;
   }
-  
+
+  if (action == 'auto-repay') {
+    if (network == 'test') {
+      throw new Error(
+        'repay from collateral only works on the mainnet due to low liquidity in pools'
+      );
+    }
+
+    if (privateKeyRequired) {
+      pk = params[7];
+      if (!pk) {
+        console.error('Missing private key');
+        return;
+      }
+      kit.addAccount(pk);
+    }
+
+    if (!isValidAsset(params[2])) return;
+    if (!isValidAsset(params[3])) return;
+    if (!isValidRateMode(params[4])) return;
+    if (!isNumeric(params[5])) return;
+    if (!isValidBoolean(params[6])) return;
+
+    const caller = params[0];
+    const user = params[1];
+    const collateralAsset = tokens[params[2]];
+    const debtAsset = tokens[params[3]];
+    const rateMode = params[4] === 'stable' ? 1 : 2;
+    const repayAmount = BN(web3.utils.toWei(params[5]));
+    const useFlashloan = params[6] == 'true' ? true : false;
+    const useATokenAsFrom = params[2] != 'celo';
+    const useATokenAsTo = params[3] != 'celo';
+
+    const reserveCollateralToken = await dataProvider.methods
+      .getReserveTokensAddresses(collateralAsset.options.address)
+      .call();
+    const mToken = new eth.Contract(MToken, reserveCollateralToken.aTokenAddress);
+    const reserveDebtToken = await dataProvider.methods
+      .getReserveTokensAddresses(debtAsset.options.address)
+      .call();
+
+    let maxCollateralAmount = 0;
+    if (collateralAsset != debtAsset) {
+      const uniswap = new kit.web3.eth.Contract(
+        Uniswap,
+        '0xe3d8bd6aed4f159bc8000a9cd47cffdb95f96121'
+      );
+      const amountOut = useFlashloan
+        ? repayAmount.plus(repayAmount.multipliedBy(9).dividedBy(10000))
+        : repayAmount;
+      const amounts = await uniswap.methods
+        .getAmountsIn(amountOut, [
+          useATokenAsFrom ? reserveCollateralToken.aTokenAddress : collateralAsset.options.address,
+          useATokenAsTo ? reserveDebtToken.aTokenAddress : debtAsset.options.address,
+        ])
+        .call();
+      maxCollateralAmount = BN(amounts[0])
+        .plus(BN(amounts[0]).multipliedBy(1).dividedBy(1000))
+        .toFixed(0); // 0.1% slippage
+    }
+    const feeAmount = BN(maxCollateralAmount).multipliedBy(10).dividedBy(10000);
+
+    console.log(`Checking mToken ${mToken.options.address} for approval`);
+    if (
+      BN(await mToken.methods.allowance(user, autoRepay.options.address).call()).lt(
```

Comment:
Maybe send approve tx if there is not enough allowance, just do the same approve like here in other actions

Reply:
We can't approve because we don't have the user's private key.